### PR TITLE
Allow requesting a list to lookup plugins instead of , delimited string

### LINKED ISSUE #10291: Lookup plugins cannot return list to Jinja2 template
##### Issue Type:

Bug
##### Ansible Version:

ansible 1.8.2
(Also tested by @bcoca on current devel)
##### Environment:

N/A
##### Summary:

If a lookup plugin returns a list `['one', 'two', 'three']` this is stringified and returned to a Jinja2 template as a string `one,two,three`.
###### environment

```
export ANSIBLE_LOOKUP_PLUGINS=`pwd`
```
###### lookup mini.py

``` python
from ansible import utils, errors
import os

class LookupModule(object):

    def __init__(self, basedir=None, **kwargs):
        self.basedir = basedir

    def run(self, terms, inject=None, **kwargs):

        ret = ['gato', 'chat', 'katze', 'miau' ]

        return ret
```
###### playbook

``` yaml
- hosts:
   - localhost
  gather_facts: False
  connection: local
  tasks:
  - action: template src=t.in dest=/tmp/t.out
```
###### template

```
#-- result from lookup()
{% for k in lookup('mini', 'blah')  %}
        {{ k }}
{% endfor %}

#-- result from lookup() | list
{% for k in lookup('mini', 'blah') | list  %}
        {{ k }}
{% endfor %}

#-- lookup().split()
{{ lookup('mini', 'blah').split(',')  }}

#-- Jinja2 list
{% for k in  ['uno', 'dos', 'tres' ] %}
        {{ k }}
{% endfor %}
```
##### Expected Results:

I expected the following from `lookup('mini', 'blah')`

```
    gato
    chat
    katze
    miau
```
##### Actual Results:

Results from running above playbook:

```
#-- result from lookup()
        g
        a
        t
        o
        ,
        c
        h
        a
        t
        ,
        k
        a
        t
        z
        e
        ,
        m
        i
        a
        u

#-- result from lookup() | list
        g
        a
        t
        o
        ,
        c
        h
        a
        t
        ,
        k
        a
        t
        z
        e
        ,
        m
        i
        a
        u

#-- lookup().split()
['gato', 'chat', 'katze', 'miau']

#-- Jinja2 list
        uno
        dos
        tres
```

### PR DESCRIPTION
**Pull request by bcoca**
so now you can do something like:

```
{% for k in lookup('items',"['one','two','three']",wantlist=True) %}
{{k}}
{% endfor %}
```

should fix #10291

### CODE REVIEW

File: lib/ansible/utils/template.py [hunk 1]
```diff
@@ -217,7 +218,7 @@ def template_from_file(basedir, path, vars, vault_password=None):
 
     def my_lookup(*args, **kwargs):
         kwargs['vars'] = vars
-        return lookup(*args, basedir=basedir, **kwargs)
+        return lookup( *args, basedir=basedir, **kwargs)
```
Comment (sivel) [inline lib/ansible/utils/template.py]: Is the added whitespace here on purpose?

File: lib/ansible/utils/template.py [hunk 2]
```diff
@@ -87,6 +87,7 @@ class Flags:
 def lookup(name, *args, **kwargs):
     from ansible import utils
     instance = utils.plugins.lookup_loader.get(name.lower(), basedir=kwargs.get('basedir',None))
+    wantlist = kwargs.get('wantlist', False)
```
Comment (sivel) [inline lib/ansible/utils/template.py]: Would it be beneficial to pop this off as to not pass it into `.run()`?

**Comment by sivel** [comment]
+1, I like this approach better than some of the other recommendations.

File: lib/ansible/utils/template.py (hunk 1 above)
Comment (bcoca) [inline lib/ansible/utils/template.py]: nope

File: lib/ansible/utils/template.py (hunk 2 above)
Comment (bcoca) [inline lib/ansible/utils/template.py]: yes

**Comment by jpmens** [comment]
@bcoca `'msg': "UnboundLocalError: local variable 'wantlist' referenced before assignment", 'failed': True}`

**Comment by jpmens** [comment]
Other than 

`wantlist = False`
before

``` python
if 'wantlist' in kwargs:
```

SHIP IT!

;-)

**Comment by sivel** [comment]
might be easier to use:

```
wantlist = kwargs.pop('wantlist', False)
```

**Comment by bcoca** [comment]
i did not know pop could also use the same default values as get ... something new every day

**Comment by jpmens** [comment]
Thank you, Brian!

**Pull request merged**